### ISSUE TITLE
Snapshots not created

### ISSUE THREAD

**Issue by jinsumv** [fastlane/snapshot#356]
Thanks for creating the snapshot tool. I'm trying to take some screenshots using the application.I followed the instructions in the Github page. In my Swift UI test case, I'm invoking the snapShot() function. However, the snapshot tool does not create any snapshots. 

I read through the "how it works" section and I looked at /tmp/snapshot_derived and I cannot see any snapshots created by Xcode. 

 I'm using:

Xcode Version 7.1.1 (7B1005)
OSX Yosemite 10.10.5 (14F27)
Testing on iPhone 6s simulator

How can I debug this further?

**Comment by jinsumv** [comment]
For what it's worth, snapshot is running xcodebuild as:

xcodebuild -workspace './MyApp_iOS.xcworkspace' -scheme 'iPhoneAutomation' -derivedDataPath '/tmp/snapshot_derived/' -destination 'platform=iOS Simulator,id=07E84B67-0005-4BB9-B39F-2114529ACB55,OS=9.1' build test | tee '/Users/jinsumv/Library/Logs/snapshot/Video-iPhoneAutomation.log'

The compilation succeeded and artifacts like the compiled object files went into /tmp/snapshot_derived/. Then the automation tests started running, but the screenshots are not getting saved at all. I would appreciate any help in debugging this.

BTW I have a question - does xcodebuild save the screenshots only after finishing all test cases or does it create the screenshots as it runs the tests one by one?

**Comment by ohwutup** [comment]
hey @jinsumv would you mind sharing your UI test code? it would help to see how you have the snapshots set up.

**Comment by jinsumv** [comment]
Thanks @ohwutup. Its working now.
But the issue was: I have more than 100 test cases. If any one of them fails, its not saving the screenshot. It saves screenshot only if all the tests pass. Is there any way to get screenshot of passing tests?

**Comment by lilidotshi** [comment]
+1 Also it would be nice to have a retry function. Sometimes network can be unpredictable with downloading things, something like --max-retries 3 for any failing test.

**Comment by ohwutup** [comment]
currently, the features are implemented such that screenshots are saved upon a successful full test run.  i will leave this issue open for more discussion and it would be awesome if someone would want to submit a PR to add in this functionality.

@lilidotshi if i understand your question correctly, it seems as though that is a different feature request than what this issue was created about. would be great to discuss that on another thread if you wouldn't mind starting a new issue?

**Comment by fastlanebot** [comment]
This issue was migrated to https://github.com/fastlane/fastlane/issues/2522. Please post all further comments there.

`fastlane` is now a mono repo, you can read more about the change in our [blog post](https://krausefx.com/blog/our-goal-to-unify-fastlane-tools). All tools are now available in the [fastlane main repo](https://github.com/fastlane/fastlane) :rocket:
